File: projects/OMSATrackingAPI/OMSATrackingAPI.DAL/Models/Driver.cs
namespace OMSATrackingAPI.DAL.Models
{
    public class Driver : BaseEntity<int>
    {
        public string Name { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string IndentificationDocument { get; set; } = string.Empty;
        public int BusId { get; set; }

        // Navigation properties
        public Bus Bus { get; set; } = null!;
    }
}

File: projects/omsa_tracking_app/lib/models/driver.dart
class Driver {
  String name;
  String lastName;
  String identificationDocument;
  int busId;

  Driver({
    required this.name,
    required this.lastName,
    required this.identificationDocument,
    required this.busId,
  });

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      name: json['name'],
      lastName: json['lastName'],
      identificationDocument: json['identificationDocument'],
      busId: json['busId'],
    );
  }
}
